bouncer/blacklist/views.py

```diff
@@ -1,6 +1,40 @@
 from django.shortcuts import render
 from django.http import HttpResponse
+from django.views import View
+from django.template import loader
+from blacklist import models
+from blacklist.brain import (
+    is_ip_blacklisted,
+    is_email_blacklisted,
+    is_email_host_blacklisted,
+)
 
-# Create your views here.
-def index(request):
-    return HttpResponse("Blacklist")
+
+class RequestView(View):
+    template = loader.get_template("blacklist/ip_request.html")
+
+    def get(self, request, *args, **kwargs):
+        context = {"request_list": []}
+        if "email" in self.request.GET:
+            context["request_list"].append("email")
```

Comment:
`request_list` is a bit misleading for what it actually does, I would name it something like `results`.

Also, the current implementation, while it works, uses an 1-dimension array for the results, which is not very Pythonic.

I would change it to a list of tuples, ie`[(kind, value, result), ...]`. This would make the template even more readable.